### ISSUE TITLE
Didn't catch an unsupported torch:copy / wrong gradient computation

### ISSUE THREAD

**Issue by as1986** [twitter/torch-autograd#106]
When I do this:
```lua
params = {W={torch.DoubleTensor(4,2):fill(2),torch.DoubleTensor(4,2):fill(3)}}
inp = torch.DoubleTensor(4):fill(1)

my_func = function(params, inp)
    local s = params.W[1].new(params.W[1]:size(1), params.W[1]:size(2))
    torch.copy(s, params.W[1])
    return torch.sum(s) + torch.sum(params.W[2]) + torch.sum(inp)
end
df = autograd(my_func, {optimize=false})
grad, loss =df(params, z)
```

I get grad.W[1] as a zero matrix (should be a matrix filled with 1). I think torch.copy is not yet supported but this is somehow not caught.

**Comment by alexbw** [comment]
`torch.copy` is actually overloaded to produce a return value. So, try `s = torch.copy(s, params.W[1])`

**Comment by alexbw** [comment]
Yep, returning the output of `torch.copy` in s works for me.

```
autograd = require 'autograd'
torch = require 'torch'

params = {W={torch.DoubleTensor(4,2):fill(2),torch.DoubleTensor(4,2):fill(3)}}
inp = torch.DoubleTensor(4):fill(1)

my_func = function(params, inp)
    local s = params.W[1].new(params.W[1]:size(1), params.W[1]:size(2))
    s = torch.copy(s, params.W[1]) -- this is the only change!
    return torch.sum(s) + torch.sum(params.W[2]) + torch.sum(inp)
end
df = autograd(my_func, {optimize=false})
grad, loss = df(params, inp)
print(grad.W[1]) -- prints a matrix with 1s
```

In general, in-place operations will not work in autograd, always try to grab the return value.

**Comment by as1986** [comment]
Thanks!
